Src/DependencyCollector/Shared/Implementation/HttpProcessing.cs

```diff
@@ -0,0 +1,404 @@
+namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+    using System.Linq;
+    using System.Net;
+#if !NET40
+    using System.Web;
+#endif
+    using Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Common;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.DependencyCollector.Implementation.Operation;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+#if NET40
+    using Microsoft.ApplicationInsights.Net40;
+#endif
+    using Microsoft.ApplicationInsights.Web.Implementation;
+
+    /// <summary>
+    /// Concrete class with all processing logic to generate RDD data from the callbacks
+    /// received from Profiler instrumentation for HTTP or HTTP EventSource/DiagnosticSource events.   
+    /// </summary>
+    internal abstract class HttpProcessing
```

Comment:
>abstract [](start = 13, length = 8)

we found this generalization a bad idea. Abstract class starts knowing all sorts of internal details of children classed. In this code review - Profiler and Diagnostics Source may share headers reading and injection logic as well as appending VERB to the name. So perhaps should share the base class as well. Also adding new collectors becoming a very hard task as you may need to refactor how the base class works. Consider adding helper classes instead of the base class.